jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/FilesTaskV2.java

```diff
@@ -0,0 +1,191 @@
+/*
+ * Copyright 2019 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.gradle;
+
+import com.google.cloud.tools.jib.plugins.common.SkaffoldFilesOutput;
+import com.google.common.base.Preconditions;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.Project;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.artifacts.Dependency;
+import org.gradle.api.artifacts.ProjectDependency;
+import org.gradle.api.artifacts.PublishArtifact;
+import org.gradle.api.initialization.Settings;
+import org.gradle.api.plugins.JavaPluginConvention;
+import org.gradle.api.tasks.SourceSet;
+import org.gradle.api.tasks.TaskAction;
+
+/**
+ * Prints out changing source dependencies on a project.
+ *
+ * <p>Expected use: {@code ./gradlew _jibSkaffoldFiles -q} or {@code ./gradlew
+ * :<subproject>:_jibSkaffoldFiles -q}
+ */
+public class FilesTaskV2 extends DefaultTask {
+
+  private final SkaffoldFilesOutput skaffoldFilesOutput = new SkaffoldFilesOutput();
```

Comment:
I think we should not have the task be stateful like this, since if `listFiles` is called twice there would be two JSONs printed out.

Reply:
I'm not sure I understand, this task will only be called once. The gradle system wont add it to the task tree twice?

Reply:
Ah okay, it just felt like the variable could be local to the `listFiles` method to avoid any potential misuse of `listFiles` by calling it more than once.